src/TriggerBinding/SqlTableChangeMonitor.cs

```diff
@@ -370,11 +375,9 @@ private async void RunLeaseRenewalLoopAsync()
                     this._logger.LogDebugWithThreadId("BEGIN OpenLeaseRenewalLoopConnection");
                     await connection.OpenAsync(token);
                     this._logger.LogDebugWithThreadId("END OpenLeaseRenewalLoopConnection");
+
                     while (!token.IsCancellationRequested)
                     {
-                        this._logger.LogDebugWithThreadId("BEGIN WaitRowsLock - LeaseRenewal");
-                        await this._rowsLock.WaitAsync(token);
-                        this._logger.LogDebugWithThreadId("END WaitRowsLock - LeaseRenewal");
```

Comment:
Waiting on semaphore inside `RenewLeasesAsync` so we have same method where we are waiting and releasing the semaphore.